ksqldb-common/src/main/java/io/confluent/ksql/name/ColumnNames.java

```diff
@@ -71,41 +93,52 @@ public static boolean isAggregate(final ColumnName name) {
     return name.text().startsWith(AGGREGATE_COLUMN_PREFIX);
   }
 
-  /**
-   * Determines the next unique column alias.
-   *
-   * <p>Finds any existing {@code KSQL_COL_x} column names in the supplied {@code sourceSchema} to
-   * ensure the returned generated column name is unique.
-   *
-   * @param sourceSchema the source schema.
-   * @return a column name in the form {@code KSQL_COL_x} which does not clash with source schema.
-   */
-  public static ColumnName nextGeneratedColumnAlias(final LogicalSchema sourceSchema) {
-    final int maxExistingIdx = maxGeneratedAliasIndex(sourceSchema);
-    return generatedColumnAlias(maxExistingIdx + 1);
+  private static OptionalInt extractGeneratedAliasIndex(final ColumnName columnName) {
+    final Matcher matcher = GENERATED_ALIAS_PATTERN.matcher(columnName.text());
+    return matcher.matches()
+        ? OptionalInt.of(Integer.parseInt(matcher.group(1)))
+        : OptionalInt.empty();
   }
 
-  /**
-   * Determines the highest index of generated column names like {@code KSQL_COL_x} in the supplied
-   * {@code sourceSchema}.
-   *
-   * @param sourceSchema the schema.
-   * @return the highest index or {@code -1}
-   */
-  private static int maxGeneratedAliasIndex(final LogicalSchema sourceSchema) {
-    return sourceSchema.columns().stream()
+  private static IntStream generatedAliasIndexes(final Stream<LogicalSchema> sourceSchema) {
```

Comment:
Is this a `Stream` here to handle joins where we will have multiple source schemas?

Reply:
Correct. Joins can have multiple sources...